nexustreetojson.py

```diff
@@ -12,30 +25,59 @@ def _is_array(possible_array):
 
 
 def _attrs_to_json(tree):
-    names = sorted(tree.attrs)
+    """
+    Creates a list of lines of JSON for the attributes of a given group or dataset
+
+    :param tree: Group or dataset tree node
+    :return: List of lines of JSON
+    """
     result = []
-    for k in names:
-        if nexusformat.nexus.tree.is_text(tree.attrs[k]):
-            attr_string = u'": "' + nexusformat.nexus.tree.text(tree.attrs[k]) + '"'
-        elif _is_array(tree.attrs[k]):
-            array_string = np.array2string(tree.attrs[k], separator=', ').replace(' ', '')
-            array_string = array_string.replace('.,', '.0,')
-            array_string = array_string.replace('.]', '.0]')
-            attr_string = u'": ' + array_string
-        else:
-            attr_string = u'": ' + nexusformat.nexus.tree.text(tree.attrs[k])
-        txt = (u'{"' + k + attr_string).replace("u'", "'")
-        txt += '}'
-        result.append(txt)
-    result[0] = '\"attributes\": [' + result[0]
-    result[-1] += ']'
-    return ', '.join(result)
+    if tree.attrs:
+        for k in tree.attrs:
+            if nexusformat.nexus.tree.is_text(tree.attrs[k]):
+                attr_string = u'": "' + nexusformat.nexus.tree.text(tree.attrs[k]) + '"'
+            elif _is_array(tree.attrs[k]):
+                array_string = np.array2string(tree.attrs[k], separator=', ').replace(' ', '')
+                array_string = array_string.replace('.,', '.0,')
+                array_string = array_string.replace('.]', '.0]')
+                attr_string = u'": ' + array_string
+            else:
+                attr_string = u'": ' + nexusformat.nexus.tree.text(tree.attrs[k])
+            txt = (u'"' + k + attr_string).replace("u'", "'")
+            result.append(txt)
+    if tree.nxclass != "NXfield":
+        result.append('"NX_class": "' + tree.nxclass + '"')
+    if result:
+        result[0] = '\"attributes\": {' + result[0]
+        result[-1] += '}'
+        return_lines = ', '.join(result)
```

Comment:
Is there some specific reason why you build up the json return value via string concatenation?

Reply:
I think creating a python dictionary is a better approach.
Originally I was hoping I could just tweak the tree print output from nexusformat, that's one reason I ended up doing it with strings. The other is that I thought it would be the most straightforward way of outputting the fields in a nice order, `name` first for example.

Reply:
There is the possiblity to use `OrderedDict` together with `json.dumps` to preserve the order.